sdk/core/azure-core/azure/core/rest/_helpers.py

```diff
@@ -0,0 +1,279 @@
+# --------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# The MIT License (MIT)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+import os
+import codecs
+from enum import Enum
+from json import dumps
+import collections
+from typing import (
+    Optional,
+    Union,
+    Mapping,
+    Sequence,
+    List,
+    Tuple,
+    IO,
+    Any,
+    Dict,
+    Iterable,
+    Iterator,
+    cast,
+    Callable,
+)
+import xml.etree.ElementTree as ET
```

Comment:
If I remember correctly, xml.etree is very heavy. We can consider import it only needed.

Reply:
ok sounds good

Reply:
actually @xiangyan99 , we end up having to use ET whenever people pass in `content` to the `HttpRequest` (and it has to be the first check bc it's the most specific), so I think I'm going to leave it at the top of the module, since I think it becomes more costly if we have to import it everytime someone passes in `content`. Thanks!